File: Section1 - Basics/Patterns/pattern8.js
// BruteForce O(n^2)
function printTriangle(n) {
	let str = "";
	for (let i = n; i > 0; i--) {
		for (let j = i; j < n; j++) {
			str += " ";
		}
		for (let k = 0; k < i - 1; k++) {
			str += "*";
		}
		for (let l = 0; l < i; l++) {
			str += "*";
		}
		str += "\n";
	}
	console.log(str);
}

printTriangle(5);

// Optimised O(n^2)
function printTriangle(n) {
	let str = "";
	for (let i = 1; i <= n; i++) {
		for (let j = 1; j < i; j++) {
			str += " ";
		}
		for (let k = 1; k <= 2 * (n - i) + 1; k++) {
			str += "*";
		}
		str += "\n";
	}
	console.log(str);
}

printTriangle(4);

File: Section1 - Basics/Patterns/pattern8.java
// BruteForce O(n^2)
void printTriangle(int n) {
    // code here
    for(int i=n; i>0; i--) {
        for(int j=i; j<n; j++) {
            System.out.print(" ");
        }
        for(int j=0; j<i-1; j++) {
            System.out.print("*");
        }
        for(int j=0; j<i; j++) {
            System.out.print("*");
        }
        System.out.print("\n");
    }
}

// Striver
void printTriangle(int n) {
    // code here
    for (int i=0; i<n; i++) {
        for(int j=0; j<i; j++) {
            System.out.print(" ");
        }
        for(int k=0; k<2*(n-i)-1; k++) {
            System.out.print("*");
        }
        for(int l=0; l<i; l++) {
            System.out.print(" ");
        }
    System.out.print("\n");
    }
}

// Optimised O(n^2)
void printTriangle(int n) {
    // code here
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j < i; j++) {
            System.out.print(" ");
        }
        for (int k = 1; k <= 2 * (n - i) + 1; k++) {
            System.out.print("*");
        }
        System.out.println();
    }
}